src/libs/registry/docset.cpp

```diff
@@ -607,3 +617,183 @@ QString Docset::parseSymbolType(const QString &str)
 
     return aliases.value(str, str);
 }
+
+// ported from DevDocs' searcher.coffee:
+// (https://github.com/Thibaut/devdocs/blob/50f583246d5fbd92be7b71a50bfa56cf4e239c14/assets/javascripts/app/searcher.coffee#L91)
+static void matchFuzzy(
+    int nLen, const unsigned char *needle, int hLen, const unsigned char *haystack,
+    int *start, int *len
+) {
+    int j = 0, groups = 0;
+    for (int i = 0; i < nLen; ++i) {
+        bool found = false, first = true;
+        int distance = 0;
+        while (j < hLen) {
+            bool match = needle[i] == haystack[j++];
+            if (match) {
```

Comment:
`match` is not used anywhere else, just use condition in `if`. Also simply `++j` one line above.